src/modules/top-level.nix

```diff
@@ -192,22 +227,24 @@ in
       mkdir -p .devenv
       rm -f .devenv/profile
       ln -s ${profile} .devenv/profile
+      unset ${lib.concatStringsSep " " config.unsetEnvVars}
     '';
 
     shell = performAssertions (
-      mkNakedShell {
+      pkgs.mkShell ({
         name = "devenv-shell";
-        env = config.env;
-        profile = profile;
-        shellHook = config.enterShell;
-        debug = config.devenv.debug;
-      }
+        packages = [ profile ];
+        shellHook = ''
```

Comment:
I wonder if we could evaluate the nix flake again here, because `$PWD` is accessible in `shellHook`.

We can pass `$PWD` to the nix flake like this:
```
nix eval .#devShells.${system}.default.passthru.evalWithRoot --arg "$PWD"
```

Reply:
Sure `PWD` is accessible, but it's going to be where you happen to run `nix develop` from, not the root of your project directory.  You could probably search upwards for the root though, in the same way nix commands do (i.e. search for the nearest `nix flake.nix`).